### PR TITLE
Add support for getting latest release information.

### PR DESCRIPTION
**Pull request by archit**
Needed to add exception for `latest` release.

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -1856,10 +1856,10 @@ def get_releases(self):
     def get_release(self, id):
         """
         :calls: `GET /repos/:owner/:repo/releases/:id https://developer.github.com/v3/repos/releases/#get-a-single-release
-        :param id: int (release id), str (tag name)
+        :param id: int (release id), str (tag name), latest (latest release)
```
Comment (juharris) [inline github/Repository.py]: This makes it look like latest is a type.  How about "... str (tag name or "latest")"?

File: github/Repository.py [hunk 2]
```diff
@@ -1856,10 +1856,10 @@ def get_releases(self):
...
         :rtype: None or :class:`github.GitRelease.GitRelease`
         """
-        if isinstance(id, int):
+        if isinstance(id, int) or id is "latest":
```
Comment (juharris) [inline github/Repository.py]: `==` is preferred over `is` for strings

**Comment by archit** [comment]
Ready for review!

**Comment by jzelinskie** [comment]
What if there's a tag named "latest"? Even though it's a terrible tag name, it is possible.

**Comment by archit** [comment]
@jzelinskie that is a good question. I'm not really sure what the answer is, or if GitHub even accounts for that. Do you think it would be better to setup a whole new method instead? `Repository#get_latest_release` ?

**Comment by jzelinskie** [comment]
Yep, I think that sounds like a good solution.

**Comment by archit** [comment]
Ready for review.

**Comment by jzelinskie** [comment]
Looks like it's failing to build on Travis -- I'm not sure why those checks aren't showing up in the PR.

https://travis-ci.org/PyGithub/PyGithub/jobs/80322806

**Comment by archit** [comment]
whats the best way to rebuild  in travis?

**Comment by jzelinskie** [comment]
When you add a new commit, it will rebuild.

**Comment by yannispanousis** [comment]
+1

**Comment by archit** [comment]
I had to close & reopen, to trigger a build with travis CI.

**Comment by jakirkham** [comment]
This seems really neat. Is there anything we can do to help get this in?

**Comment by jzelinskie** [comment]
When this passes CI, I'll merge it.

**Comment by sfdye** [comment]
Close in favor of #609

**Pull request closed without merging**